drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/upload_clients_job.rb

```diff
@@ -8,36 +8,131 @@
 module HmisExternalApis::AcHmis
   class UploadClientsJob < BaseJob
     include NotifierConfig
+    include Rails.application.routes.url_helpers
 
     attr_accessor :state
 
-    def perform
+    def perform(mode = 'clients_with_mci_ids_and_address')
       setup_notifier('HMIS Upload Clients')
-
       if Exporters::ClientExportUploader.can_run?
-        export = Exporters::ClientExport.new
-        export.run!
-
-        uploader = Exporters::ClientExportUploader.new(
-          io_streams: [
-            OpenStruct.new(
-              name: 'Client.csv',
-              io: export.output,
-            ),
-          ],
-        )
-
-        uploader.run!
+        case mode
+        when 'clients_with_mci_ids_and_address' then clients_with_mci_ids_and_address
+        when 'hmis_csv_export' then hmis_csv_export
+        when 'project_crosswalk' then project_crosswalk
+        else
+          raise "invalid item to upload: #{mode}"
+        end
         self.state = :success
       else
         self.state = :not_run
-        Rails.logger.info 'Not running client upload due to lack of credentials'
+        Rails.logger.info "Not running #{mode} due to lack of credentials"
       end
     rescue StandardError => e
       puts e.message
       self.state = :failed
       @notifier.ping('Failure in upload clients job', { exception: e })
       Rails.logger.fatal e.message
     end
+
+    private
+
+    def clients_with_mci_ids_and_address
+      export = Exporters::ClientExport.new
+      export.run!
+
+      uploader = Exporters::ClientExportUploader.new(
+        filename_format: '%Y-%m-%d-clients.zip',
+        io_streams: [
+          OpenStruct.new(
+            name: 'Client.csv',
+            io: export.output,
+          ),
+        ],
+      )
+
+      uploader.run!
+    end
+
+    def hmis_csv_export
+      # FIXME: Not sure what this should be a hash of. I'll refactor to take content
+      # into account in a later step if that's needed. Could be a hash of the
+      # current time or content or sorted list of unique ids in the file. Or,
+      # maybe something else.
+      hash = SecureRandom.hex(8)
+
+      # FIXME: what user should I use? Can I make exports not need a user?
+      user = User.first
```

Comment:
```suggestion
      user = Hmis::Hud::User.system_user(data_source_id: HmisExternalApis::AcHmis.data_source.id)
```
You'll probably want to move this below the `data_source =` below.